client/src/components/Related/RelatedProducts.jsx

```diff
@@ -43,6 +54,29 @@ const RelatedProducts = (props) => {
           : null
         }
       </div>
+      {modalShowing ?
+        <div className='comparison-modal'>
+          <div className='modal-top'>COMPARING</div>
+          <div className='modal-product-names'>
+            <div className='product-1'>{currentProduct.name}</div>
+            <div className='product-2'>{comparisonProduct.name}</div>
+          </div>
+          <table className='modal-table'>
+            <tbody>
+              {getFeatures().map((feature, index) => {
+                return (
+                  <tr key={`${feature}-${index}`}>
+                    <td className='left-check'>{currentProduct.features.filter(item => item.feature === feature.feature && item.value === feature.value).length > 0 ? '✓' : null}</td>
+                    <td>{feature.value} {feature.feature}</td>
+                    <td className='right-check'>{comparisonProduct.features.filter(item => item.feature === feature.feature && item.value === feature.value).length > 0 ? '✓' : null}</td>
+                  </tr>
+                );
+              })}
+            </tbody>
+          </table>
+          <div className="close-btn" onClick={() => { setModalShowing(false); }}></div>
+        </div>
+        : null}
```

Comment:
This all looks really great. Love the modal pop up. One thing to consider is with all of your conditional rendering, what if you put everything highlighted here in it's own component? It could then look more like: `modalShowing ? < ModalPopUp /> : null` It's not critical, but could make it easier for yourself to see what all is happening especially if you did this wherever you have conditional rendering.

Reply:
I think that's a great idea that I should consider for future refactoring, thank you!